google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -30,28 +30,49 @@ class _MetadataService:
     def __init__(self):
         self._experiment = None
         self._run = None
+        self._metric = None
```

Comment:
do you need to track _metric name here? You are generating it deterministically. Maybe just have a private method to retrieve the metric name as used it wherever needed

Reply:
yes, we need this as we are creating the metric artifact in set_run now and in log_metrics, we need to get the artifact by its resource name.